2748. Number of Beautiful Pairs/solution.py

```diff
@@ -1,31 +1,52 @@
 class Solution:
-    # This takes in a list of nums and returns each beautiful pair,
-    # where a beautiful pair is when the first digit of the first
+    # This takes in a list of ints and returns the number of beautiful pairs.
+    # A beautiful pair is when the first digit of the first
     # num in the pair is coprime with the last digit of the second
-    # num in the pair. Two nums are coprime if their gcd is no greater
-    # than 1.
+    # num in the pair. Two nums are coprime if their gcd is 1.
     def countBeautifulPairs(self, nums: List[int]) -> int:
         # the number of beautiful pairs
+        beautifulPairs = 0
         
-        # convert each int in the list to strings to save
-        # time from converting each int to a string inside the 
-        # main loop
+        # convert each int in the list to tuples of the first
+        # and last digit to save conversion time
+        for i in range(len(nums)):
+            num = nums[i]
+            numStr = str(num)
+            nums[i] = (int(numStr[0]), int(numStr[-1]))
+        
+        # print(nums)
         
         # iterate to the second-to-last digit of nums
+        for i in range(len(nums) - 1):
+            num1Tuple = nums[i]
         
-            # iterate through the next nums
+            # iterate from next element in nums to end of nums
+            for j in range(i + 1, len(nums)):
+                num2Tuple = nums[j]
+                
+                # print(num1Tuple, num2Tuple)
             
                 # find the first char of the first digit
+                num1 = num1Tuple[0]
                 
-                # find last char of second digit (convert both to int)
+                # find last char of second digit
+                num2 = num2Tuple[1]
                 
-                # iterate backwards from 9 to 2
+                # assume that the pair is coprime unless proven otherwise,
+                # makes the code simpler
+                beautifulPairs += 1
                 
-                    # if both ints are divisible by this number, that
-                    # means they are not coprime!
+                # print("beautifulPairs before GCD:", beautifulPairs)
                 
-                # if they are coprime, then the pair is beautiful and
-                # the count of beautiful numbers should increase
+                # iterate backwards from 9 to 2
+                for divisor in range(9, 1, -1):
```

Comment:
Encapsulation 💊!

Reply:
Hmm I'll think about that.

You are probably right actually, but it's more convenient to just do this.